R/plotGrid.R

```diff
@@ -0,0 +1,66 @@
+#' Create a plot grid
+#'
+#' @param plotGridConfiguration  A `PlotGridConfiguration` object, which
+#'   is an `R6` class object that defines properties of a plot grid (like number
+#'   of rows, columns, labels, etc.).
+#'
+#' @description
+#'
+#' Create a plot grid using the `patchwork::wrap_plots()` function. The required
+#' arguments are supplied through the `PlotGridConfiguration` object.
+#'
+#' @examples
+#'
+#' library(ospsuite)
+#' library(ggplot2)
+#' library(tlf)
+#'
+#' # plots to be arranged in a grid
+#' set.seed(123)
+#' ls_plots <- list(
+#'   tlf::plotHistogram(x = rnorm(100)),
+#'   tlf::plotHistogram(x = rnorm(100, mean = 3)),
+#'   tlf::plotHistogram(x = rnorm(100, mean = 10))
+#' )
+#'
+#' # create an instance of plot configuration class
+#' plotGridObj <- PlotGridConfiguration$new(plotList = ls_plots)
+#'
+#' # specify further customizations for the plot grid
+#' plotGridObj$title <- "my combined plot"
+#' plotGridObj$subtitle <- "something clever"
+#' plotGridObj$caption <- "my sources"
+#' plotGridObj$nColumns <- 2L
+#' plotGridObj$tagLevels <- "A"
+#' plotGridObj$tagPrefix <- "Plot ("
+#' plotGridObj$tagSuffix <- ")"
+#'
+#' # plot the grid
+#' plotGrid(plotGridObj)
+#'
+#' @references
+#' For more, see: <https://patchwork.data-imaginist.com/articles/patchwork.html>
+#'
+#' @export
+plotGrid <- function(plotGridConfiguration) {
+  patchwork::wrap_plots(
+    PlotGridConfiguration$plotList,
+    ncol = PlotGridConfiguration$nColumns,
```

Comment:
there is a mix of instance `plotGridConfiguration` and class `PlotGridConfiguration` going on here. I don't get it

Reply:
Yeah, that was unintentional and introduced in my latest commit.

I will fix it tomorrow.

Reply:
Fixed this. 

Following the coding convention-

Class name: UpperCamelCase (`PlotGridConfiguration`)
Parameter name:  lowerCamelCase (`plotGridConfiguration`)